src/chainstate/stacks/boot/pox-2.clar

```diff
@@ -261,6 +264,64 @@
+               ;; todo: assert eq
+               (cycle-entry-u (get stacker cycle-entry))
+               (cycle-entry-total-ustx (get total-ustx cycle-entry))
+               (cycle-last-entry-ix (- (get len (unwrap-panic (map-get? reward-cycle-pox-address-list-len { reward-cycle: cycle }))) u1)))
+            (asserts! (is-eq cycle-entry-u (some (get stacker data))) (err ERR_STACKING_CORRUPTED_STATE))
+            (if (not (is-eq cycle-last-entry-ix set-index))
+                ;; do a "move" if the entry to remove isn't last
+                (let ((move-entry (unwrap-panic (map-get? reward-cycle-pox-address-list { reward-cycle: cycle, index: cycle-last-entry-ix }))))
+                    (map-set reward-cycle-pox-address-list
+                             { reward-cycle: cycle, index: set-index }
+                             move-entry)
+                    (match (get stacker move-entry) moved-stacker
+                     ;; if the moved entry had an associated stacker, update its state
+                     (let ((moved-state (unwrap-panic (map-get? stacking-state { stacker: moved-stacker })))
+                           ;; calculate the index into the reward-set-indexes that `cycle` is at
+                           (moved-cycle-index (- cycle (get first-reward-cycle moved-state)))
+                           (moved-reward-list (get reward-set-indexes moved-state))
+                           ;; reward-set-indexes[moved-cycle-index] = set-index via slice, append, concat.
+                           (update-list-inner (print (concat (append (default-to (list) (slice moved-reward-list u0 moved-cycle-index))
+                                                      set-index)
+                                                      (default-to (list) (slice moved-reward-list (+ moved-cycle-index u1) (len moved-reward-list))))))
+                           (update-list (unwrap-panic (as-max-len? update-list-inner u12))))
+                          (map-set stacking-state { stacker: moved-stacker }
+                                   (merge moved-state { reward-set-indexes: update-list })))
+                     ;; otherwise, we dont need to update stacking-state after move
+                     true))
+                ;; if not moving, just noop
+                true)
+            ;; in all cases, we now need to delete the last list entry
+            (map-delete reward-cycle-pox-address-list { reward-cycle: cycle, index: cycle-last-entry-ix })
+            (map-set reward-cycle-pox-address-list-len { reward-cycle: cycle } { len: cycle-last-entry-ix })
+            ;; finally, update `reward-cycle-total-stacked`
+            (map-set reward-cycle-total-stacked { reward-cycle: cycle }
+                { total-ustx: (- (get total-ustx (unwrap-panic (map-get? reward-cycle-total-stacked { reward-cycle: cycle })))
+                                 cycle-entry-total-ustx) })
+            (ok (merge data { cycle: (+ u1 cycle)} )))))
+
```

Comment:
Maybe add a comment here indicating that this can only get called from an internal process -- i.e. as processing early unlocks?